### PR TITLE
Graphical editor docs and cleanup

### PR DESCRIPTION
**Pull request by philip-alldredge**
This includes Javadoc updates and various cleanups. The focus for this merge is the public API but other areas were touched. Some refactoring and renaming occurred to avoid some issues found during the documentation process. Please perform some additional testing to find potential issues.

### CODE REVIEW

File: ge/org.osate.ge/src/org/osate/ge/query/ExectableQuery.java [hunk 1]
```diff
@@ -25,52 +25,88 @@
...
+ * Represents a query which can be executed by @link QueryService} to retrieve results
  * @noextend This class is not intended to be subclassed by clients.
- *
+ * @since 3.0
  */
-public class StandaloneQuery {
-	private final DefaultQuery rootQuery = new RootQuery(() -> this.rootNode);
-	private BusinessObjectContext rootNode;
-	private final DefaultQuery query;
+public final class ExectableQuery<T> {
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/query/ExectableQuery.java:41]: Should this be ExecutableQuery?

Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/query/ExectableQuery.java]: Should be fixed now. Thanks for catching this. It took me a minute to figure out what your comment meant.

File: ge/org.osate.ge/src/org/osate/ge/services/DiagramExportService.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
 import org.eclipse.core.resources.IFile;
 
 /**
- * Service for exporting a diagram as an image.
+ * Service for exporting a diagram as an image. Editor implements provide this service as a global service.
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/services/DiagramExportService.java:37]: "Editor implements provide this service" needs to be reworded

**Review by RyanMcilnay — CHANGES_REQUESTED** [review]

File: ge/org.osate.ge.gef/src/org/osate/ge/gef/FeatureConstants.java [hunk 3]
```diff
@@ -27,19 +27,25 @@
...
 	 * Height of the feature.
 	 */
-	public static double HEIGHT = 16.0;
+	public static final double HEIGHT = 16.0;
 
 	/**
 	 * The smallest dimension.
 	 */
-	public static double MIN_DIMENSION = Math.min(WIDTH, HEIGHT);
+	public static final double MIN_DIMENSION = Math.min(WIDTH, HEIGHT);
```
[the remark this answers is not in the document]
Comment (philip-alldredge) [reply ge/org.osate.ge.gef/src/org/osate/ge/gef/FeatureConstants.java:50]: Practically yes. Although that could change if the constants are adjusted and using the min shouldn't cause any problems. Not sure if the compiler is smart enough to optimize it out though.  I think I'd like to leave it as it is for now. The min wasn't introduced in this branch.

File: ge/org.osate.ge/src/org/osate/ge/internal/query/DescendantsByBusinessObjectRelativeReferencesQuery.java [hunk 4]
```diff
@@ -25,61 +25,59 @@
...
+import java.util.function.Function;
 
 import org.osate.ge.BusinessObjectContext;
 import org.osate.ge.RelativeBusinessObjectReference;
 import org.osate.ge.internal.services.ReferenceService;
-import org.osate.ge.query.Supplier;
 
-public class DescendantsByBusinessObjectRelativeReferencesQuery extends DefaultQuery {
+public class DescendantsByBusinessObjectRelativeReferencesQuery<T> extends DefaultQuery<T> {
 	private final static RelativeBusinessObjectReference[] nullBoRefs = new RelativeBusinessObjectReference[0];
```
[the remark this answers is not in the document]
Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/internal/query/DescendantsByBusinessObjectRelativeReferencesQuery.java:35]: Leave as is for now. That wasn't affected by this branch and I haven't documented or reviewed it yet.

File: ge/org.osate.ge.gef/src/org/osate/ge/gef/FeatureConstants.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (RyanMcilnay) [reply ge/org.osate.ge.gef/src/org/osate/ge/gef/FeatureConstants.java:50]: Okay, works for me

File: ge/org.osate.ge/src/org/osate/ge/services/DiagramExportService.java (hunk 2 above)
Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/services/DiagramExportService.java]: Fixed.

**Review by RyanMcilnay — APPROVED** [review]

**Pull request merged**